.github/workflows/cml.yaml

```
@@ -4,20 +4,29 @@ jobs:
   run:
     runs-on: [ubuntu-latest]
     container: docker://dvcorg/cml:0-dvc2-base1
+    env:
+      GIT_USER: Olivaw[bot]
+      GIT_EMAIL: 64868532+iterative-olivaw@users.noreply.github.com
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
+          ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0
+      - name: Fix git safe.directory in container
+        run: |
+          git config --global --add safe.directory ${GITHUB_WORKSPACE}
+          git config --global user.name ${GIT_USER}
+          git config --global user.email ${GIT_EMAIL}
```

also does it make sense to add this functionality to CML @iterative/cml?

```suggestion
          cml ci
```

---

seems best /CC @0x2b3bfa0 (as per https://github.com/iterative/cml/issues/970, https://github.com/iterative/cml/issues/984)

---

https://github.com/iterative/cml/issues/1250#issuecomment-1304314803